src/instrumentation/vercel-sdk.ts

```diff
@@ -92,6 +93,22 @@ const extractTools = (options: Options<AllVercelFn>): ITool[] | undefined => {
   }));
 };
 
+const extractPrompt = (
+  options: Options<AllVercelFn>
+):
+  | { uuid: string; promptId: string; variables: Record<string, any> }
+  | undefined => {
+  if (!options.messages) return undefined;
+
+  for (const message of options.messages) {
+    if (typeof (message as any).literalMetadata === 'function') {
+      return (message as any).literalMetadata();
+    }
```

Comment:
slightly cleaner style IMO

```suggestion
    const metadata = (message as any).literalMetadata();

    if (typeof metadata === 'function') {
      return metadata();
    }
```

Reply:
the suggestion does not work if the messages are not linked to a prompt. .literalMetadata() does not exist.